### PR TITLE
Feature/sl 1488 amend session

### PR DESCRIPTION
**Pull request by radoslawlandowskikainos**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/session/AmendSessionAction.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    @Override
+    public UUID getUserTransactionId() {
+        return amendSessionRequest.getUserTransactionId();
+    }
+
+    @Override
+    public UUID[] getAssociatedEntitiesIds() {
+        return new UUID[] { amendSessionRequest.getId() };
```
Comment (r-kruk-kainos) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/session/AmendSessionAction.java]: Did we discuss if we should also check hearingParts associated to that session?

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/session/AmendSessionAction.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        return new FactMessage(RulesService.UPSERT_SESSION, msg);
+    }
+
+    @Override
+    public List<UserTransactionData> generateUserTransactionData() {
+        List<UserTransactionData> userTransactionDataList = new ArrayList<>();
+        userTransactionDataList.add(new UserTransactionData("session",
+            session.getId(),
+            currentSessionAsString,
+            "update",
```
Comment (r-kruk-kainos) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/session/AmendSessionAction.java]: We could potentially extract these "updates" and other operations maybe to action interfaces or somewhere else.

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/model/db/Session.java [hunk 3]
```diff
@@ -72,7 +72,7 @@
         })
     private SessionType sessionType;
 
-    public void setSessionType(SessionType sessionType) {
+    public void  setSessionType(SessionType sessionType) {
```
Comment (r-kruk-kainos) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/model/db/Session.java]: Unnecessary space

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/service/RevertChangesManager.java [hunk 4]
```diff
@@ -56,6 +56,25 @@ private void handleTransactionData(UserTransactionData utd) {
             String msg = factsMapper.mapDbSessionToRuleJsonMessage(session);
             rulesService.postMessage(utd.getUserTransactionId(), RulesService.DELETE_SESSION, msg);
 
+        } else if ("session".equals(utd.getEntity()) && "update".equals(utd.getCounterAction())) {
```
Comment (r-kruk-kainos) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/service/RevertChangesManager.java]: Please extract a handleSession method

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/session/AmendSessionAction.java (hunk 2 above)
Comment (radoslawlandowskikainos) [reply src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/session/AmendSessionAction.java]: Great idea for further improvements

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/session/AmendSessionAction.java (hunk 1 above)
Comment (kpakur) [reply src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/session/AmendSessionAction.java]: @r-kruk-kainos there was something but cannot find this, and not sure if this was session to hearings or rather hearings to session.

**Pull request merged**